src/platform/plugins/shared/discover/public/application/main/discover_main_route.tsx

```diff
@@ -134,14 +134,21 @@ export function DiscoverMainRoute({
           stateContainer.actions.loadDataViewList(),
         ]);
 
-        if (!hasUserDataViewValue || !defaultDataViewExists) {
+        const hasAdHocDataViews = stateContainer.internalState.getState().adHocDataViews.length > 0;
+
+        if (
+          (!hasUserDataViewValue || !defaultDataViewExists) &&
+          (!hasAdHocDataViews || !hasESDataValue)
```

Comment:
This condition is hard to read. Can we use some intermediate variables which would explain it better?